File: backend/app/models/Team.py
from typing import List
from .player import Player

class Team:
    def __init__(self, name: str):
        self.name = name
        self.players: List[Player] = []
        self.group_name: str = "" # Associated group

        self.goals = 0
        self.assists = 0
        
        # Standings fields
        self.points = 0
        self.wins = 0
        self.draws = 0
        self.losses = 0
        self.goals_for = 0
        self.goals_against = 0

    def add_player(self, player: Player):
        self.players.append(player)

    def remove_player(self, player: Player):
        self.players.remove(player)

    def team_size(self):
        return len(self.players)

    def team_goals(self):
        return sum(p.goals for p in self.players)

    def team_assists(self):
        return sum(p.assists for p in self.players)
        
    def goal_difference(self):
        return self.goals_for - self.goals_against
        
    def add_match_result(self, gf: int, ga: int):
        self.goals_for += gf
        self.goals_against += ga
        if gf > ga:
            self.wins += 1
            self.points += 3
        elif gf == ga:
            self.draws += 1
            self.points += 1
        else:
            self.losses += 1

File: Frontend/lib/models/team.dart
import 'player.dart';

class Team {
  final String name;
  final List<Player> players;
  final String groupName;
  final int goals;
  final int assists;
  
  // Standings
  final int points;
  final int wins;
  final int draws;
  final int losses;
  final int goalsFor;
  final int goalsAgainst;
  final int goalDifference;

  const Team({
    required this.name,
    required this.players,
    this.groupName = '',
    this.goals = 0,
    this.assists = 0,
    this.points = 0,
    this.wins = 0,
    this.draws = 0,
    this.losses = 0,
    this.goalsFor = 0,
    this.goalsAgainst = 0,
    this.goalDifference = 0,
  });

  factory Team.fromJson(Map<String, dynamic> json) => Team(
        name: json['name'] as String? ?? 'Unnamed Team',
        groupName: json['group_name'] as String? ?? '',
        players: (json['players'] as List<dynamic>? ?? [])
            .map((e) => Player.fromJson(e as Map<String, dynamic>))
            .toList(),
        goals: (json['goals'] as num?)?.toInt() ?? 0,
        assists: (json['assists'] as num?)?.toInt() ?? 0,
        points: (json['points'] as num?)?.toInt() ?? 0,
        wins: (json['wins'] as num?)?.toInt() ?? 0,
        draws: (json['draws'] as num?)?.toInt() ?? 0,
        losses: (json['losses'] as num?)?.toInt() ?? 0,
        goalsFor: (json['goals_for'] as num?)?.toInt() ?? 0,
        goalsAgainst: (json['goals_against'] as num?)?.toInt() ?? 0,
        goalDifference: (json['goal_difference'] as num?)?.toInt() ?? 0,
      );
}
